lib/iris/tests/integration/netcdf/test_general.py

```diff
@@ -361,5 +362,125 @@ def test_lat_not_loaded(self):
             _ = cube.coord("lat")
 
 
+@tests.skip_data
+class TestDatasetAndPathLoads(tests.IrisTest):
+    @classmethod
+    def setUpClass(cls):
+        cls.filepath = tests.get_data_path(
+            ["NetCDF", "global", "xyz_t", "GEMS_CO2_Apr2006.nc"]
+        )
+        cls.phenom_id = "Carbon Dioxide"
+        cls.expected = iris.load_cube(cls.filepath, cls.phenom_id)
+
+    def test_basic_load(self):
+        # test loading from an open Dataset, in place of a filepath spec.
+        ds = threadsafe_nc.DatasetWrapper(self.filepath)
+        result = iris.load_cube(ds, self.phenom_id)
+        ds.close()
```

Comment:
Is it worth ensuring that the dataset isn't already closed? Or does this line check this?